File: backend/src/main/java/com/example/libraryapp/domain/bookitemloan/model/BookItemLoan.java
package com.example.libraryapp.domain.bookitemloan.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@AllArgsConstructor
public class BookItemLoan {
    private Long id;
    private LocalDateTime creationDate;
    private LocalDateTime dueDate;
    private LocalDateTime returnDate;
    private BookItemLoanStatus status;
    private Long userId;
    private Long bookId;
    private Long bookItemId;
}

File: frontend/src/app/modules/catalog/shared/models/book-item-loan.ts
import { BookItemLoanStatus as LoanStatus } from "../enums/book-item-loan-status";

export class BookItemLoan {
    id: number;
    creationDate: Date;
    dueDate: Date;
    returnDate: Date;
    status: LoanStatus;
    userId: number;
    bookId: number;
    bookItemId: number;
}